### ISSUE TITLE
Can I use g++ or Intel CC to compile?

### ISSUE THREAD

**Issue by ysh329** [pytorch/extension-ffi#11]
Thanks in advance.

**Comment by apaszke** [comment]
you should be able to do this by exporting a `CC` env variable

**Comment by ysh329** [comment]
@apaszke 
Do you mean: write `export CC=icc` or `alias CC='icc'` in `~/.bashrc` file?
I'm new to configure environment variables, thanks a lot!

**Comment by soumith** [comment]
he means `export CC=icc`.
You can google for them: https://www.google.com/#safe=off&q=cc+cxx+environment+variables

**Comment by ysh329** [comment]
@soumith Thanks

**Comment by ysh329** [comment]
@soumith I `export CC=icc` in command line. Of course, I installed ICC. However, after successfully executing `build.py`, I executed `python test.py`, which printed messages below:
```Shell
Traceback (most recent call last):
  File "test.py", line 12, in <module>
    from network.pose_model import PoseModel
  File "/home/yuanshuai/code/pose_estimation.pytorch/network/pose_model.py", line 8, in <module>
    from bnn.bnn_layer import BNBinaryConv2d, BinaryConv2d, BNRealConv2d
  File "/home/yuanshuai/code/pose_estimation.pytorch/network/bnn/bnn_layer.py", line 9, in <module>
    from _ext import bnn_layer
  File "/home/yuanshuai/code/pose_estimation.pytorch/network/bnn/_ext/bnn_layer/__init__.py", line 3, in <module>
    from ._bnn_layer import lib as _lib, ffi as _ffi
ImportError: /home/yuanshuai/code/pose_estimation.pytorch/network/bnn/_ext/bnn_layer/_bnn_layer.so: undefined symbol: _intel_fast_memset
```
I can use icc to compile C/C++ programs using `intrinsics` normally and my version is below:
`icc version 16.0.1 (gcc version 4.8.0 compatibility)`

**Comment by jiseongg** [comment]
@ysh329 

Have you succeeded to build PyTorch with `icc`? I'm doing this for some purpose (https://github.com/intel/mkl-dnn/issues/549) but I failed. I used the command below.

```bash
DEBUG=1 USE_CUDA=0 CC=icc CXX=icpc python setup.py develop
```
There was many compile errors about the namespace `std`. I've been struggling with PyTorch for quite long time to find out how to succeed it. If you've solved the problem, I wanna ask you to share some information...
